addons/binding/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/handler/TeslaHandler.java

```diff
@@ -971,37 +1009,45 @@ public void run() {
                                 try {
                                     logger.debug("Event Stream : Received an event: '{}'", line);
                                     String vals[] = line.split(",");
-                                    if (Long.valueOf(vals[0]) > eventLastEventTimeStamp) {
-                                        eventLastEventTimeStamp = Long.valueOf(vals[0]);
-                                        logger.trace("Event Stream : event stamp is {}", DATE_FORMATTER.format(date));
-                                        for (int i = 0; i < EventKeys.values().length; i++) {
-                                            try {
-                                                TeslaChannelSelector selector = TeslaChannelSelector
-                                                        .getValueSelectorFromRESTID((EventKeys.values()[i]).toString());
-                                                if (!selector.isProperty()) {
-                                                    State newState = teslaChannelSelectorProxy.getState(vals[i],
-                                                            selector, editProperties());
-                                                    if (newState != null && !vals[i].equals("")) {
-                                                        updateState(selector.getChannelID(), newState);
+                                    try {
+                                        lock.lock();
+                                        if (Long.valueOf(vals[0]) > lastTimeStamp) {
+                                            lastTimeStamp = Long.valueOf(vals[0]);
+                                            date.setTime(Long.valueOf(vals[0]));
+                                            logger.trace("Event Stream : Event stamp is {}",
+                                                    DATE_FORMATTER.format(date));
+                                            for (int i = 0; i < EventKeys.values().length; i++) {
```

Comment:
the Tesla API returns a comma separated string of values, that are separated into vals[], with the order of the values equal to the EventKeys Enum order.